webapp/src/main/java/uk/gov/dvsa/motr/web/component/subscription/persistence/DynamoDbPendingSubscriptionRepository.java

```
@@ -119,12 +122,21 @@ public void delete(PendingSubscription subscription) {
         Map<String, Object> expressionAttributeValues = new HashMap<>();
         expressionAttributeValues.put(":id", subscription.getConfirmationId());
 
-        dynamoDb.getTable(tableName).deleteItem(
-                key,
-                "id = :id",
-                null,
-                expressionAttributeValues
-        );
+        try {
+            dynamoDb.getTable(tableName).deleteItem(
+                    key,
+                    "id = :id",
+                    null,
+                    expressionAttributeValues
+            );
+        } catch (AmazonDynamoDBException e) {
+            EventLogger.logEvent(new PendingSubscriptionDeletionFailedEvent()
+                    .setConfirmationId(subscription.getConfirmationId())
+                    .setVrm(subscription.getVrm())
+                    .setContact(subscription.getContact())
+                    .setMessage(e.getMessage())
+                    .setErrorMessage(e.getErrorMessage()));
```

Shouldn't we re-throw the error?

---

Done.